src/Collector/ProfileClientFactory.php

```diff
@@ -24,38 +24,17 @@ class ProfileClientFactory implements ClientFactory
      */
     private $factory;
 
-    /**
-     * @var Collector
-     */
-    private $collector;
-
-    /**
-     * @var Formatter
-     */
-    private $formatter;
-
-    /**
-     * @var Stopwatch
-     */
-    private $stopwatch;
-
     /**
      * @param ClientFactory|callable $factory
      */
-    public function __construct($factory, Collector $collector, Formatter $formatter, Stopwatch $stopwatch)
+    public function __construct($factory, private readonly Collector $collector, private readonly Formatter $formatter, private readonly Stopwatch $stopwatch)
     {
         if (!$factory instanceof ClientFactory && !is_callable($factory)) {
             throw new \RuntimeException(sprintf('First argument to ProfileClientFactory::__construct must be a "%s" or a callable.', ClientFactory::class));
```

Comment:
obsolete with the parameter type declaration
```suggestion
```

Reply:
++